File: javascript/bob/bob.js
//
// This is only a SKELETON file for the "Bob" exercise. It's been provided as a
// convenience to get you started writing code faster.
//

var Bob = function() {};

Bob.prototype.hey = function(input) {
    if (input.trim() === '') {
        return 'Fine. Be that way!';
    }

    if (input.toUpperCase() === input && input.toUpperCase() != input.toLowerCase()) {
        return 'Whoa, chill out!';
    }

    if (input.indexOf('?') == input.length - 1) {
        return 'Sure.';
    }

    return 'Whatever.';
};

module.exports = Bob;

File: fsharp/bob/Bob.fs
module Bob

open System.Text.RegularExpressions

let response (input: string): string =
    let trimmed = input.Trim()
    let isYelled = trimmed = trimmed.ToUpper() && not (trimmed = trimmed.ToLower())
    let isQuestion = trimmed.Length > 0 && trimmed.[trimmed.Length - 1] = '?'

    if trimmed.Length = 0 then "Fine. Be that way!"
    elif isYelled && isQuestion then "Calm down, I know what I'm doing!"
    elif isYelled then "Whoa, chill out!"
    elif isQuestion then "Sure."
    else "Whatever."